### PR TITLE
Added variables supporting separate additional tags for writer and reader

### PR DESCRIPTION
**Pull request by bergbrains**
## Adds ability to assign additional tags to writer and reader
* Additional tags are merged with `module.this.tags`
* Separate variables for reader and writers

## To allow observability at the db-role level
* In order to provide a way to query for the reader and writer in metrics/events/logs

### CODE REVIEW

**Comment by bergbrains** [comment]
The auto-format error is really a perms issue.

I added cloudposse bot as a contributing writer, after these initial errors.

File: variables.tf [hunk 1]
```diff
@@ -472,3 +472,15 @@ variable "intra_security_group_traffic_enabled" {
   default     = false
   description = "Whether to allow traffic between resources inside the database's security group."
 }
+
+variable "additional_primary_tags" {
```
Comment (nitrocode) [inline variables.tf:476]:
```suggestion
variable "primary_additional_tags" {
```

File: variables.tf [hunk 2]
```diff
@@ -472,3 +472,15 @@ variable "intra_security_group_traffic_enabled" {
...
+  type        = map(string)
+  description = "Map of additional tags to add to primary instance tags."
+  default     = {}
+}
+
+variable "additional_reader_tags" {
```
Comment (nitrocode) [inline variables.tf:482]:
```suggestion
variable "secondary_additional_tags" {
```

File: main.tf [hunk 3]
```diff
@@ -184,7 +186,7 @@ resource "aws_rds_cluster" "secondary" {
   db_subnet_group_name                = join("", aws_db_subnet_group.default.*.name)
   db_cluster_parameter_group_name     = join("", aws_rds_cluster_parameter_group.default.*.name)
   iam_database_authentication_enabled = var.iam_database_authentication_enabled
-  tags                                = module.this.tags
+  tags                                = local.replica_tags
```
Comment (nitrocode) [inline main.tf:189]:
```suggestion
  tags                                = merge(module.this.tags, var.secondary_additional_tags)
```

File: main.tf [hunk 4]
```diff
@@ -246,7 +248,7 @@ resource "aws_rds_cluster_instance" "default" {
   db_subnet_group_name                  = join("", aws_db_subnet_group.default.*.name)
   db_parameter_group_name               = join("", aws_db_parameter_group.default.*.name)
   publicly_accessible                   = var.publicly_accessible
-  tags                                  = module.this.tags
+  tags                                  = local.primary_tags
```
Comment (nitrocode) [inline main.tf:251]:
```suggestion
  tags                                = merge(module.this.tags, var.primary_additional_tags)
```

File: main.tf [hunk 5]
```diff
@@ -7,6 +7,8 @@ locals {
   is_regional_cluster      = var.cluster_type == "regional"
   is_serverless            = var.engine_mode == "serverless"
   ignore_admin_credentials = var.replication_source_identifier != "" || var.snapshot_identifier != null
+  primary_tags             = merge(module.this.tags, var.additional_primary_tags)
+  replica_tags             = merge(module.this.tags, var.additional_reader_tags)
```
Comment (nitrocode) [inline main.tf:11]:
```suggestion
```

**Comment by nitrocode** [comment]
Let's make the names consistent with this PR https://github.com/cloudposse/terraform-aws-alb/pull/122/files

**Comment by nitrocode** [comment]
@bergbrains please run `make pr/auto-format` locally to fix the auto format errors

**Comment by bergbrains** [comment]
Maybe i'll get to this again someday.

**Pull request closed without merging**